e2e/test/scenarios/embedding/embedding-reproductions.cy.spec.js

```diff
@@ -504,106 +500,179 @@ describe("issues 20845, 25031", () => {
+          cy.wrap(card_id).as("questionId");
+
+          const mapFilterToCard = {
+            parameter_mappings: [
+              {
+                parameter_id: dashboardParameter.id,
+                card_id,
+                target: ["dimension", ["template-tag", TEMPLATE_TAG_NAME]],
+              },
             ],
-            parameters: [DASHBOARD_PARAMETER],
           };
-          return createDashboardWithTabs(dashboardDetails);
-        })
-        .then(dashboard => {
-          cy.wrap(dashboard.id).as("dashboardId");
+
+          cy.editDashboardCard(dashboardCard, mapFilterToCard);
+
+          cy.request("POST", `/api/dashboard/${dashboard_id}/public_link`).then(
+            ({ body: { uuid } }) => {
+              cy.wrap(uuid).as("dashboardPublicUuid");
+            },
+          );
         });
     });
-  });
 
-  it("should allow dashboard filter to map to boolean field filter parameter in static embedding (metabase#27643)", () => {
-    cy.log("Test dashboard");
-    cy.get("@dashboardId").then(dashboardId => {
-      visitDashboard(dashboardId);
+    it("should allow a dashboard filter to map to a boolean field filter parameter in static embedding (metabase#27643)", () => {
+      cy.log("Test the dashboard");
+      cy.get("@dashboardId").then(dashboardId => {
+        visitDashboard(dashboardId);
+      });
+
+      toggleFilterWidgetValues(["false"]);
+      getDashboardCard().findByText("Rows 1-4 of 455").should("be.visible");
+
+      cy.log("Test the embedded dashboard");
+      cy.get("@dashboardId").then(dashboard => {
+        visitEmbeddedPage({
+          resource: { dashboard },
+          params: {},
+        });
+      });
+
+      toggleFilterWidgetValues(["false"]);
+      getDashboardCard().findByText("Rows 1-4 of 455").should("be.visible");
+
+      cy.log("Test the public dashboard");
+      cy.signOut();
+      cy.get("@dashboardPublicUuid").then(uuid => {
+        cy.visit(`/public/dashboard/${uuid}`);
+      });
+      toggleFilterWidgetValues(["false"]);
+      getDashboardCard().findByText("Rows 1-4 of 455").should("be.visible");
     });
+  });
 
-    toggleFilterWidgetValues(["false"]);
-    getDashboardCard().findByText("Rows 1-6 of 455").should("be.visible");
+  describe("questions", () => {
+    beforeEach(() => {
+      cy.get("@postgresInvoicesExpectedInvoiceId").then(fieldId => {
+        createNativeQuestion(getQuestionDetails(fieldId), {
+          wrapId: true,
+          idAlias: "questionId",
+        });
+      });
 
-    cy.log("Test embedded dashboard");
-    cy.get("@dashboardId").then(dashboardId => {
-      visitEmbeddedPage({
-        resource: { dashboard: dashboardId },
-        params: {},
+      cy.get("@questionId").then(questionId => {
+        cy.request("POST", `/api/card/${questionId}/public_link`).then(
+          ({ body: { uuid } }) => {
+            cy.wrap(uuid).as("questionPublicUuid");
+          },
+        );
```

Comment:
The same comment as for the publicDashboard. There's `visitPublicQuestion` helper that is doing all this for you.